### PR TITLE
feat: use atomic increment for kvcounter

### LINKED ISSUE #229: Use the atomic wasi:keyvalue/atomic interface for WIT-ified kvcounter
Since the rust host has [partial support for the `wasi:keyvalue/atomic` interface](https://github.com/wasmCloud/wasmCloud/pull/555) now, we should update the example `kvcounter-wasmcon2023` actor to use it.

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

This adds use of `wasi:keyvalue/atomic` for doing atomic increments in the kvcounter example 

Resolves #229 

## Testing

Built on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

Tested on platform(s)
- [x] x86_64-linux
- [ ] aarch64-linux
- [ ] x86_64-darwin
- [ ] aarch64-darwin
- [ ] x86_64-windows

### Manual Verification

Tested locally with the new Rust host (`wash` > v0.19)

The instructions show the old UI, but doing all setup with `wash` CLI works

### CODE REVIEW

**Review by brooksmtownsend — APPROVED** [review]
Nice!

**Comment by vados-cosmonic** [comment]
Thanks for taking a look @brooksmtownsend  -- I can't merge this so if you wouldn't mind

**Pull request merged**
